azure-pipelines-build.yml

```diff
@@ -0,0 +1,114 @@
+parameters:
+  - name: buildAPI
+    displayName: Build API
+    type: boolean
+    default: true
+  - name: buildWeb
+    displayName: Build Web
+    type: boolean
+    default: true
+
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      endpoint: Planning-Inspectorate
+      name: Planning-Inspectorate/common-pipeline-templates
+      ref: feature/multi-job-ci
+
+extends:
+  template: stages/wrapper_ci.yml@templates
+  parameters:
+    globalVariables:
+      - template: azure-pipelines-variables.yml@self
+    validationJobs:
+      - name: Run Linting & Tests
+        steps:
+          - checkout: self
+            clean: true
+            persistCredentials: true
+          - template: ../steps/check_branch_name.yml
+          - template: ../steps/azure_auth.yml
+            parameters:
+              subscriptionId: $(SUBSCRIPTION_ID)
+          - template: ../steps/node_script.yml
+            parameters:
+              script: npm ci
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run tscheck
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run lint:js
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run test
+          - template: ../steps/check_changed_files.yml
+            parameters:
+              developmentBranch: develop
+              pathFilters:
+                - name: packages
+                  path: packages
+                - name: web
+                  path: apps/web
+                - name: api
+                  path: apps/api
+              stepName: check_files
+      - name: Build Web
```

Comment:
I notice that on the jobs for this PR, the lint tasks have failed, but the build tasks ran anyway? Shouldn't they have aborted?

Reply:
Which one are ya looking at? This run they skipped - https://dev.azure.com/planninginspectorate/back-office/_build/results?buildId=30429&view=results.

In the "condition" attribute I've included the `succeeded()` condition. So they should only run on success. It might be because you were looking at one I ran while testing I hope!